pkg/operator/vspherecontroller/checks/check_migration_disabled.go

```diff
@@ -0,0 +1,33 @@
+package checks
+
+import (
+	"context"
+	"fmt"
+
+	operatorv1 "github.com/openshift/api/operator/v1"
+)
+
+const (
+	storageOperatorName = "cluster"
+)
+
+type CheckMigrationDisabled struct{}
+
+func (c *CheckMigrationDisabled) Check(ctx context.Context, checkOpts CheckArgs) []ClusterCheckResult {
+	storageOperator, err := checkOpts.apiClient.GetStorage(storageOperatorName)
+	if err != nil {
+		checkResult := ClusterCheckResult{
+			CheckStatus: CheckStatusOpenshiftAPIError,
+			CheckError:  err,
+			Action:      CheckActionDegrade,
+			Reason:      fmt.Sprintf("failed to check Storage operator %s: %v", storageOperatorName, err),
+		}
+		return []ClusterCheckResult{checkResult}
+	}
+	// Migration is disabled unless this is explicitly set to CSIWithMigrationDriver
+	if storageOperator.Spec.VSphereStorageDriver != operatorv1.CSIWithMigrationDriver {
```

Comment:
IIRC we wanted `Upgradeable=true` when `VSphereStorageDriver==""` ("I don't care").

Reply:
Good catch, updated